app/Http/Controllers/Api/ChannelController.php

```diff
@@ -0,0 +1,298 @@
+     */
+    public function postAChannel(Request $request)
+    {
+        $userId = 1;// The id of the user gotten from the token
+
+        if ($this->validateUserRequestForEmptyFields($request)) {
+            return Response::json(['message' => 'All fields are required'], 400);
+
+        }
+
+        if ($this->validateUserRequest($request)) {
+            return Response::json(['message' => 'Channel already exists'], 400);
+
+        } 
+
+        $channel = Channel::create([
+            'channel_name' => $request->input('channel_name'),
+            'channel_description' => $request->input('channel_description'),
+            'created_at' => date('Y-m-d h:i:s'),
+            'user_id' => $userId,
+        ]);
+
+        if (count($channel) > 0) {
+            return Response::json(['message' => 'Channel created successfully'], 201);
+        }
+    }
+
+    /**
+     * This method updates a channel using PUT verb
+     *
+     * @param $channel_name
+     * @param $channel_description
+     *
+     * @return $response
+     */
+    public function editAChannel(Request $request, $channel_name)
+    {
+        if ($this->validateUserRequestForEmptyFields($request)) {
+            return Response::json(['message' => 'All fields are required'], 400);
+
+        }
+
+        $channel = Channel::where('channel_name', '=', $channel_name)
+        ->orWhere('channel_name', '=', strtolower($channel_name))
+        ->first();
+
+        if ($this->processEditChannel($request, $channel)) {
+            return Response::json(['message' => 'Channel updated successfully'], 200);
+
+        }
+
+        return Response::json([
+            'message' => 'Channel cannot be updated because the channel name is incorrect'
+        ], 404);
+
+    }
+
+    /**
+     * This method updates a channel using PATCH verb
+     *
+     * @param $channel_name
+     *
+     * @return $response
+     */
+    public function editASingleChannelResource(Request $request, $channel_name)
+    {
+        if ($this->validateUserRequestForPatchRequest($request)) {
+            return Response::json(['message' => 'All fields are required'], 400);
+
+        }
+
+        $channel = Channel::where('channel_name', '=', $channel_name)
+        ->orWhere('channel_name', '=', strtolower($channel_name))
+        ->first();
+
+        if ($this->processEditChannelForPatchRequest($request, $channel)) {
+            return Response::json(['message' => 'Channel updated successfully'], 200);
+
+        }
+
+        return Response::json([
+            'message' => 'Channel cannot be updated because the channel name is incorrect'
+        ], 404);
+
+    }
+
+    /**
+     * This method completes the processing of editing user using PUT verb
+     *
+     * @param $request
+     * @param $channel
+     *
+     * @return $response
+     */
+    public function processEditChannel($request, $channel)
+    {
+        if (! is_null($channel)) {
+            DB::table('channels')
```

Comment:
**Code Quality - Nitpick**
@andela-tolotin For the sake of consistency can we stick to using Eloquent and not Query Builder. Please fix this all through your code.